idioms/coercion-arguments.md

```diff
@@ -0,0 +1,115 @@
+# Use borrowed types for arguments
+
+## Description
+
+Using a target of a deref coercion can increase the flexibility of your code when you are deciding which argument type to use for a function argument.
+In this way, the function will accept more input types.
+
+For example; using `&str` instead of a `&String`, or `&[T]` in preference of `&Vec<T>`.  This is not limited to slice-able or fat pointer types. In fact you should always prefer using the __borrowed type__ over __borrowing the owned type__, e.g., `&T` to `&Box<T>`.
```

Comment:
```suggestion
For example; using `&str` instead of a `&String`, or `&Path` instead of `&PathBuf`, or `&[T]` in preference of `&Vec<T>`.  This is not limited to slice-able or fat pointer types. In fact you should always prefer using the __borrowed type__ over __borrowing the owned type__, e.g., `&T` to `&Box<T>`.
```

Reply:
I'm not sure if we should state that without discussing the general differences of `Path` and `PathBuf`, I feel the example without stating it is much clearer for the comparisons.